ports/mimxrt/mimxrt_flash.c

```diff
@@ -153,3 +155,37 @@ MP_DEFINE_CONST_OBJ_TYPE(
     make_new, mimxrt_flash_make_new,
     locals_dict, &mimxrt_flash_locals_dict
     );
+
+#if MICROPY_VFS_ROM
+
+#ifndef MICROPY_HW_ROMFS_BYTES
+#define MICROPY_HW_ROMFS_BYTES (256 * 1024)
+#endif
+// Put ROMFS at the upper end of the respective code space.
+// For machine.mem32 the absolute address is required, for the flash functions
+// erase and write the offset to the flash start address.
+#define MICROPY_HW_ROMFS_ADDRESS (((uint32_t)&__vfs_start) - MICROPY_HW_ROMFS_BYTES)
```

Comment:
Should this be `__vfs_end-MICROPY_HW_ROMFS_BYTES`? Also I think `#define MICROPY_HW_FLASH_STORAGE_BYTES (((uint32_t)&__vfs_end) - ((uint32_t)&__vfs_start))` in `flash.h` isn't right (storage bytes would be less if there's a rom image).

Perhaps using the linker script to specify where the image is placed would be less confusing? It would also decouple it from `vfs.
```
#if MICROPY_VFS_ROM
_micropy_hw_romfs_start = MICROPY_HW_ROMFS_BASE;
_micropy_hw_romfs_end =   MICROPY_HW_ROMFS_BASE + MICROPY_HW_ROMFS_LENGTH;
#endif
```
If not, can we make this configurable to allow boards to specify where the romfs image is placed?
```C
#ifndef MICROPY_HW_ROMFS_BASE
....
```

Reply:
The VfsRom file system is intentionally placed before the vfs file system, reducing the text segment. So it does not affect the vfs file system. When making the commit I considered adding separate base & size definitions for romfs, but did not follow that idea in favor of  smaller change. 
About defining these places per board: Since the values have to be known by the linker to be sure that they match other areas in flash, they have to be defined in the Make file. That happens already for the total size of the flash and the romfs size.

Reply:
FWIW I tested the following and it works fine:
```C
extern uint8_t _micropy_hw_romfs_start;
extern uint8_t _micropy_hw_romfs_end;

#define MICROPY_HW_ROMFS_BASE   ((uintptr_t)&_micropy_hw_romfs_start - (uintptr_t)&__flash_start) 
#define MICROPY_HW_ROMFS_BYTES  ((uintptr_t)&_micropy_hw_romfs_end - (uintptr_t)&_micropy_hw_romfs_start)
....
        case 1: // address
            return mp_obj_new_int((uintptr_t)&_micropy_hw_romfs_start);
```

Reply:
Sure that works. It's the same pattern as for vfs. Only _micropy_hw_romfs_start and _micropy_hw_romfs_end have to be defined in the loader file.

Reply:
> Sure that works. It's the same pattern as for vfs. Only _micropy_hw_romfs_start and _micropy_hw_romfs_end have to be defined in the loader file.

Is this a problem or can we use them? I think this would be more flexible, for example I would like to place romfs after flash filesystem (I use a custom, common, linker script to define the above symbols).

Reply:
At the moment the linker defined vfs_start and vfs_end. So I would prefer romfs_start and romfs_end. Or vfsrom_start and vfsrom_end.

Reply:
No problem, I can define multiple symbols.

Reply:
Excellent! Thanks!

Reply:
I have pulled in the latest commits from @robert-hh 's branch, including the change mentioned above.